microsoft/vscode
# support switching workspace storage for web

This PR enables us to switch the workspace storage for web storage. This was already implemented for the native storage service, but the web storage service threw an error whenever it was called.

Let me know if this is a wanted PR, or if this doesn't align with the roadmap of VSCode. I'm looking to use this function, so I thought it wouldn't hurt to upstream the changes.

## Testing

1. Start a workspace
2. Call `storageService.switch(/* new workspace */, true)`
3. Verify that new data is stored in the new IndexedDB entry & retrieved when loading that workspace
